### PR TITLE
fix: AJDA-2621 authenticationMechanism for default mongodb:// protocol

### PR DESCRIPTION
**Pull request by MiroCillik**
## Summary

Phase 2 follow-up to PR #35 (AJDA-2621 / SUPPORT-15991). Phase 1 migrated `testConnection()` to `mongosh` so the UI's "Test Connection" button works against Azure Cosmos DB. Phase 2 fixes the actual extraction path — `mongoexport` on the default `mongodb://` protocol previously omitted `--authenticationMechanism`, falling back to SCRAM-SHA-1, which Cosmos DB rejects.

- Adds optional `db.authenticationMechanism` enum (`SCRAM-SHA-1`, `SCRAM-SHA-256`, `MONGODB-X509`, `GSSAPI`, `PLAIN`, `MONGODB-AWS`); rejected under `custom_uri` (users embed `authMechanism=` in the URI directly).
- Default `mongodb://` flag-based path now emits `--authenticationMechanism <value>` after `--authenticationDatabase`.
- `UriFactory` appends `authMechanism=<value>` query param for `mongodb` and `mongodb+srv` protocols (used by `mongoexport --uri` for SRV and by `mongosh` for `testConnection`).
- Existing code path with `authenticationMechanism` unset emits byte-identical commands to before.

Customer impact: once released, CSAS can switch their config from `custom_uri` workaround back to default `mongodb://` protocol with `authenticationMechanism: SCRAM-SHA-256`.

## Test plan

- [x] Unit tests — 118 pass (added 10 new cases across `ConfigDefinitionTest`, `UriFactoryTest`, `ExportCommandFactoryTest`)
- [x] Functional tests — 80 pass (added new `export-with-auth-mechanism` scenario running SCRAM-SHA-256 against `mongodb-auth` service)
- [x] `composer ci` (validate + phplint + phpcs + phpstan level 8 + tests) — all green
- [ ] Customer verification on `connection.cs.keboola.cloud` project 383 once released — switch to default protocol with `authenticationMechanism: SCRAM-SHA-256` and run extraction job

## Notes

- UI schema (separate platform repo) needs a matching field for end users to set this without editing raw JSON.
- Default behavior unchanged — field is opt-in. Defaulting to SCRAM-SHA-256 was rejected (would break MongoDB <4.0).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by ondrajodas — APPROVED** [review]

**Comment by MiroCillik** [comment]
@keboola-pr-reviewer review

File: src/Config/DbNode.php [hunk 1]
```diff
@@ -94,6 +103,9 @@ protected function init(NodeBuilder $builder): void
             ->scalarNode('port')->cannotBeEmpty()->end()
             ->scalarNode('database')->cannotBeEmpty()->end()
             ->scalarNode('authenticationDatabase')->end()
+            ->enumNode('authenticationMechanism')
```
[the remark this answers is not in the document]
Comment (MiroCillik) [reply src/Config/DbNode.php:117]: Fixed in b12fc9b — added `beforeNormalization()` in `DbNode` that unsets `authenticationMechanism` when it's empty/whitespace, so the enum no longer rejects `""` from the UI. Also added regression tests in `ConfigDefinitionTest` for empty values in both old + row config.

File: src/UriFactory.php [hunk 2]
```diff
@@ -75,6 +75,12 @@ private function fromParams(string $protocol, array $params): Uri
             $query[] = ['authSource', $params['authenticationDatabase']];
         }
 
+        if (isset($params['authenticationMechanism'])
+            && trim((string) $params['authenticationMechanism']) !== ''
+        ) {
+            $query[] = ['authMechanism', $params['authenticationMechanism']];
```
[the remark this answers is not in the document]
Comment (MiroCillik) [reply src/UriFactory.php:81]: Added `tests/functional/test-connection-action-with-auth-mechanism/` in b12fc9b — exercises the mongosh path through `Extractor::testConnection()` with `SCRAM-SHA-256` against `mongodb-auth`.
